test/integration/apiserver/field_validation_test.go

```diff
@@ -2887,6 +2917,103 @@ func testFieldValidationApplyUpdateCRDSchemaless(t *testing.T, rest rest.Interfa
 	}
 }
 
+func testFinalizerValidationApplyCreateAndUpdateCRD(t *testing.T, rest rest.Interface, gvk schema.GroupVersionKind, gvr schema.GroupVersionResource) {
+	var testcases = []struct {
+		name                   string
+		finalizer              string
+		updated_finalizer      string
```

Comment:
need to be able to exercise creating with multiple finalizers and updating leaving an existing invalid finalizer in place, so make these []string:

```suggestion
		finalizers              []string
		updated_finalizers      []string
```